keras/engine/topology.py

```diff
@@ -1314,13 +1334,40 @@ def get_output_shape_for(self, input_shape):
         elif self.mode == 'cos':
             return (input_shapes[0][0], 1)
 
-    def compute_mask(self, input, mask=None):
-        '''TODO: add mask merging support
-        '''
-        if mask is not None and any(mask):
-            raise Exception('Merge does not support masking, ' +
-                            'but was passed an input mask: ' + str(mask))
-        return None
+    def _normalize_mask_dims(self, mask, input):
```

Comment:
As far as I understand this method is unnecessary. However it is also undocumented, so my understanding might be incomplete.